src/core/mac/sub_mac.cpp

```diff
@@ -254,43 +253,27 @@ Error SubMac::Receive(uint8_t aChannel)
 }
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-Error SubMac::CslSample(uint8_t aPanChannel)
+void SubMac::CslSample(void)
 {
-    Error error = kErrorNone;
-
-    if (!IsCslChannelSpecified())
-    {
-        mCslChannel = aPanChannel;
-    }
-
 #if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
-    VerifyOrExit(!mRadioFilterEnabled, error = Get<Radio>().Sleep());
+    VerifyOrExit(!mRadioFilterEnabled, IgnoreError(Get<Radio>().Sleep()));
 #endif
 
-    switch (mCslState)
+    SetState(kStateCslSample);
+
+    if (mIsCslSampling && !RadioSupportsReceiveTiming())
+    {
+        IgnoreError(Get<Radio>().Receive(mCslChannel));
+    }
+    else
     {
-    case kCslSample:
-        error = Get<Radio>().Receive(mCslChannel);
-        break;
-    case kCslSleep:
 #if !OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
-        error = Get<Radio>().Sleep(); // Don't actually sleep for debugging
+        IgnoreError(Get<Radio>().Sleep()); // Don't actually sleep for debugging
 #endif
-        break;
-    case kCslIdle:
-        ExitNow(error = kErrorInvalidState);
-    default:
-        OT_ASSERT(false);
     }
 
-    SetState(kStateCslSample);
-
 exit:
```

Comment:
I guess we may get an "unused exit label" error/warning under some config combinations.
We use it only under `OPENTHREAD_CONFIG_MAC_FILTER_ENABLE` . 

We can add `#if` around the label, but better maybe to use `ExitNow()` from the `if`. Something like:
```c++
void SubMac::CslSample(void)
{
#if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
    VerifyOrExit(!mRadioFilterEnabled, IgnoreError(Get<Radio>().Sleep()));
#endif

    SetState(kStateCslSample);

    if (mIsCslSampling && !RadioSupportsReceiveTiming())
    {
        IgnoreError(Get<Radio>().Receive(mCslChannel));
        ExiNow();
    }

#if !OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
    IgnoreError(Get<Radio>().Sleep()); // Don't actually sleep for debugging
#endif

exit:
    return;
}
```
Maybe we can change this to

Reply:
Done.